src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts

```diff
@@ -378,7 +380,7 @@ export class RgwUserFormComponent extends CdForm implements OnInit {
       // Add
       // Create an observable to add the capability when the form is submitted.
       this.submitObservables.push(this.rgwUserService.addCapability(uid, cap.type, cap.perm));
-      this.capabilities.push(cap);
+      this.capabilities = [...this.capabilities, cap]; // Notify Angular CD
```

Comment:
Angular CD has not detected changes to the array after pushing new capabilities. That's because pushing an element to a list does not modify the array that way that CD gets notified. To get it working the array must be re-created.

Reply:
The datatable is already using the [OnPush strategy](https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts#L43).